backend/middleware/accesslog/accesslog.go

```
@@ -0,0 +1,91 @@
+package log
+
+// <!-- START clutchdoc -->
+// description: Enables access log filtering on responses to filter logs by status code
+// <!-- END clutchdoc -->
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/uber-go/tally"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	accesslogv1 "github.com/lyft/clutch/backend/api/config/middleware/accesslog/v1"
+	"github.com/lyft/clutch/backend/gateway/log"
+	"github.com/lyft/clutch/backend/gateway/meta"
+	"github.com/lyft/clutch/backend/middleware"
+)
+
+const Name = "clutch.middleware.accesslog"
+
+func New(config *accesslogv1.Config, logger *zap.Logger, scope tally.Scope) (middleware.Middleware, error) {
+	var statusCodes []codes.Code
+	// if no filter is provided default to logging all status codes
+	if config != nil && len(config.StatusCodeFilters) > 0 {
+		for _, filter := range config.StatusCodeFilters {
+			switch t := filter.GetFilterType().(type) {
+			case *accesslogv1.Config_StatusCodeFilter_Equals:
+				statusCode := filter.GetEquals()
+				statusCodes = append(statusCodes, codes.Code(statusCode))
+			default:
+				logger.Error("status code filter not supported", zap.String("type", fmt.Sprintf("%T", t)))
```

this should return an error